spark/src/main/scala/com/cloudera/livy/spark/SparkManager.scala

```diff
@@ -91,9 +101,25 @@ private class SparkYarnManager(processFactory: SparkProcessBuilderFactory) exten
 
   val interactiveManager = new SessionManager(processFactory.livyConf, interactiveFactory)
 
+  def clientManager: SessionManager[ClientSession, CreateClientRequest] = {
+    throw new UnsupportedOperationException
+  }
+
   override def shutdown(): Unit = {
     batchManager.shutdown()
     interactiveManager.shutdown()
     client.close()
   }
-}
\ No newline at end of file
+}
+
+private class SparkClientManager(processFactory: SparkProcessBuilderFactory)
```

Comment:
So this is needed to simply be able to create `ClientSession`s. The `SparkManager` type is specified by the configuration and only one of those is loaded by Livy. In this case, we can have 2 different `SparkClientManager`s (one for YARN and one for non-YARN), or add support for the creating `ClientSessionManager`s in both the other `SparkManager`s (which I don't mind either)